components/layouts/dashboard/dashboardDrawer/dashboardDrawerLayout/DashboardDrawerLayout.tsx

```diff
@@ -0,0 +1,66 @@
+import styles from './DashboardDrawerLayout.module.css';
+import {
+    Drawer, IconButton, List,
+} from "@mui/material";
+import React, {useEffect} from "react";
+import {bindMenu, bindToggle, usePopupState} from "material-ui-popup-state/hooks";
+import {ShowChart, ChevronRight, ChevronLeft, Group, AccessTime, Assignment, Search, Star, ViewStream, AssignmentInd} from "@mui/icons-material"
+import DashboardDrawerItem from "../dashboardDrawerItem/DashboardDrawerItem";
+
+export interface IDashboardDrawerLayout {
+    mainLayoutSetDrawerOpen: Function;
+}
+
+
+const DashboardDrawerLayout: React.FC<IDashboardDrawerLayout> = ({mainLayoutSetDrawerOpen}) => {
+
+    const DashboardDrawerPopupState = usePopupState({ variant: 'popover', popupId: 'dashboardDrawer' })
+    const DataDropDownPopupState = usePopupState({ variant: 'popover', popupId: 'dashboardDrawer' })
+
+    useEffect(() => {
+
+        mainLayoutSetDrawerOpen(DashboardDrawerPopupState.isOpen);
```

Comment:
Maybe later on we can re work how your passing around states it works for now but its a bit convolutived

Reply:
It's not that convoluted. I make a PopupState, and then pass it around to whatever needs it. There really isn't another way to do it, and I think this [library](https://github.com/jcoreio/material-ui-popup-state) is nice.

Reply:
The state change should be handled in the parent object and then passed in as a handler instead [https://stackoverflow.com/questions/35537229/how-can-i-update-the-parents-state-in-react](url) in addition there is typescript notation to better define the function that should be passed [https://www.typescriptlang.org/docs/handbook/2/functions.html](url)

Reply:
Added typescript to define the function